### PR TITLE
Enhancement/azure prefix

### PR DESCRIPTION
**Pull request by alinaciobanca**
Azure:
- enhanced the deploy_vm flow to avoid duplicate vm names
- renamed the vm_name input to vm_name_prefix
- added new flow output: vm_name

Signed-off-by: Alina Ciobanca <alina.ciobanca@hpe.com>

### CODE REVIEW

File: content/io/cloudslang/microsoft/azure/deploy_vm.sl [hunk 1]
```diff
@@ -179,8 +182,8 @@ flow:
           - auth_token
           - error_message: ${exception}
         navigate:
-          - SUCCESS: create_public_ip
           - FAILURE: on_failure
+          - SUCCESS: random_number_generator
```
Comment (victorursan) [inline content/io/cloudslang/microsoft/azure/deploy_vm.sl]: SUCCESS should be first, them FAILURE

File: content/io/cloudslang/microsoft/azure/deploy_vm.sl [hunk 2]
```diff
@@ -121,7 +124,7 @@ flow:
         default: 'https://sts.windows.net/common'
         required: false
     - location
-    - vm_name
+    - vm_name_prefix
```
Comment (victorursan) [inline content/io/cloudslang/microsoft/azure/deploy_vm.sl]: so vm_name_prefix is the 'actual' vm_name :laughing:

File: content/io/cloudslang/microsoft/azure/deploy_vm.sl (hunk 1 above)
Comment (alinaciobanca) [inline content/io/cloudslang/microsoft/azure/deploy_vm.sl]: fixed

File: content/io/cloudslang/microsoft/azure/deploy_vm.sl [hunk 3]
```diff
@@ -182,8 +182,9 @@ flow:
           - auth_token
           - error_message: ${exception}
         navigate:
-          - FAILURE: on_failure
           - SUCCESS: random_number_generator
+          - FAILURE: on_failure
+
```
Comment (victorursan) [inline content/io/cloudslang/microsoft/azure/deploy_vm.sl]: I don't think we need 2 spaces

File: content/io/cloudslang/microsoft/azure/deploy_vm.sl (hunk 2 above)
Comment (alinaciobanca) [inline content/io/cloudslang/microsoft/azure/deploy_vm.sl]: The input named vm_name_prefix is the name chosen by the user for his vm.
The actual name of the newly created vm is given by the output vm_name.

**Pull request merged**